.ci/visual-regression-test.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+# Variables
+BUILD_DIR=$(pwd)
+GITHUB_API_URL="https://api.github.com/repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME"
+
+# Check if we are NOT on the master branch and this is a PR
+if [[ ${CURRENT_BRANCH} != "master" && -z ${CI_PR_URL} ]];
+then
+  echo -e "\nVisual regression tests will only run if we are not on the master branch and making a pull request"
+  exit 0;
+fi
+
+echo -e "\nProcessing pull request #$PR_NUMBER"
+
+LAST_GIT_COMMIT_MESSAGE=$(git log -1 --pretty=%B)
+
+GIT_FILE_MODIFIED()
+{
+    # Stash list of changed files
+    GIT_FILES_CHANGED="$(git diff master --name-only)"
+
+    while read -r changedFile; do
+        if [[ "${changedFile}" == "$1" ]]
+        then
+            return 0;
+        fi
+    done <<< "$GIT_FILES_CHANGED"
+
+    return 1;
+}
+
+# Always run visual tests if "visual regression test" is in the last commit message
+if [[ ${LAST_GIT_COMMIT_MESSAGE} != *"visual regression test"* ]]
```

Comment:
I'd make the special string something like "[vr]". Existing example of a special string is "[ci-skip]" (recognized by most CI services).

Reply:
That is a good suggestion. Square brackets are special characters in Bash but I think they can just be escaped with a backslash.